SwiftGit2/Repository.swift

```diff
@@ -32,6 +73,37 @@ final public class Repository {
 		let repository = Repository(pointer)
 		return Result.Success(repository)
 	}
+
+	/// Clone the repository from a given URL.
+	///
+	/// remoteURL        - The URL of the remote repository
+	/// localURL         - The URL to clone the remote repository into
+	/// localClone       - Will not bypass the git-aware transport, even if remote is local.
+	/// bare             - Clone remote as a bare repository.
+	/// credentials      - Credentials to be used when connecting to the remote.
+	/// checkoutStrategy - The checkout strategy to use, if being checked out.
+	/// checkoutProgress - A block that's called with the progress of the checkout.
+	///
+	/// Returns a `Result` with a `Repository` or an error.
+	class public func cloneFromURL(remoteURL: NSURL, toURL: NSURL, localClone: Bool = false, bare: Bool = false,
+		credentials: Credentials = .Default(), checkoutStrategy: CheckoutStrategy = .Safe, checkoutProgress: CheckoutProgressBlock? = nil) -> Result<Repository, NSError> {
+			var checkoutOptions = SG2CheckoutOptions(checkoutProgress)
+
+			checkoutOptions.checkout_strategy = checkoutStrategy.git_checkout_strategy.rawValue
```

Comment:
Could we pass `checkoutStrategy` into `SG2CheckoutOptions()`? I'd like to encapsulate the configuration of the libgit2 structs.